### PR TITLE
Re-work install script and composer file to support minimal project structure

### PR DESCRIPTION
**Pull request by mortenson**
This PR makes a number of changes to how new Bene sites are created and initialized. The major changes are:

1. The composer.json and .gitignore files have been matched with https://github.com/drupal-composer/drupal-project. composer.lock has been removed as is the norm with other Drupal distribution composer projects.
1. The instructions for creating a new Bene site have been updated to use `composer create-project`. This matches how other Drupal distribution install instructions work and simplifies some of the install routines.
1. Initial installs now properly initialize settings.php with our standard Bene site setup (from the upstream), previously the Bene profile would never be installed when using the install script.

### CODE REVIEW

**Comment by mortenson** [comment]
@unclegcb I was also wondering if we could tie to releases in composer.json, but didn't make that change here because I wasn't sure. Ref: https://github.com/thinkshout/bene-project/blob/master/composer.json#L39

File: composer.json [hunk 1]
```diff
@@ -82,7 +82,7 @@
                 "type:bower-asset"
             ],
             "web/modules/contrib/{$name}": ["type:drupal-module"],
-            "web/profiles/{$name}": ["type:drupal-profile"],
+            "web/profiles/contrib/{$name}": ["type:drupal-profile"],
```
Comment (unclegcb) [inline composer.json]: Existing sites all skip the "contrib" subdirectory. I assume converting them will be mostly painless.

File: scripts/install.sh [hunk 2]
```diff
@@ -4,17 +4,23 @@
...
 
 INPUT_COLOR="\033[1;36m"
-MSG_COLOR="\033[0;31m"
+MSG_COLOR="\033[0;32m"
+ERROR_COLOR="\033[0;31m"
 NO_COLOR="\033[0m"
 
+# Ensure the working directory is the project root directory.
+
+cd "${0%/*}/../"
```
Comment (unclegcb) [inline scripts/install.sh]: Wow what??

Comment (mortenson) [reply scripts/install.sh]: Go up one directory from the script, basically. This addresses the case where someone would execute the script from an unexpected directory.

File: composer.json (hunk 1 above)
Comment (mortenson) [reply composer.json]: They'll all need a cache rebuild to detect the new path, but I don't expect other problems.

**Pull request merged**